app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -165,16 +199,77 @@ class SecureMessageService @Inject()(
     participants.participants.map(p1 => p1.identifier)
 
-  private def getParticipantId(client: String, conversationId: String, enrolments: Enrolments)(
-    implicit ec: ExecutionContext): Future[Option[Int]] =
+  private def getParticipant(client: String, conversationId: String, enrolments: Enrolments)(
+    implicit ec: ExecutionContext): Future[Option[Participant]] =
     repo.getConversationParticipants(client, conversationId).map {
       case Some(participants) =>
         val filteredParticipants = enrolmentMaybeParticpant(enrolments, participants).map(identifier =>
           participants.participants.filter(p => p.identifier.equals(identifier)))
-        val firstMatching = filteredParticipants.flatMap(x => x.headOption)
-        firstMatching.map(f => f.id)
+        filteredParticipants.flatMap(x => x.headOption)
+      case _ => None
+    }
+
+  private def getCaseworkerParticipant(client: String, conversationId: String, caseworkerIdentifier: SystemIdentifier)(
+    implicit ec: ExecutionContext): Future[Option[Participant]] = {
+    val identifier = Identifier(caseworkerIdentifier.name, caseworkerIdentifier.value, None)
+    repo.getConversationParticipants(client, conversationId).map {
+      case Some(participants) =>
+        participants.participants.filter(p => p.identifier.equals(identifier)).headOption.map { participant =>
+          participant
+        }
       case _ => None
     }
+  }
+
+  private def messageContentCheck(content: String): Future[Either[SecureMessageError, Boolean]] =
+    if (Base64.isBase64(content)) {
+      if (HtmlValidator.isValidHtml(content)) {
+        Future.successful(Right[SecureMessageError, Boolean](true))
+      } else {
+        Future.successful(Left[SecureMessageError, Boolean](InvalidHtmlContent("Not valid html content")))
+      }
+    } else {
+      Future.successful(Left[SecureMessageError, Boolean](InvalidBase64Content("Not valid base64 content")))
+    }
+
+  private def createMessageAndSendEmail(
+    client: String,
+    conversationId: String,
+    messagesRequest: Message,
+    participant: Participant)(
+    implicit ec: ExecutionContext,
+    headerCarrier: HeaderCarrier): Future[Either[SecureMessageError, Unit]] =
+    repo.getConversationParticipants(client, conversationId).flatMap {
+      case Some(participants) =>
+        repo.addMessageToConversation(client, conversationId, messagesRequest).flatMap {
+          case true =>
+            getEmailRequestForMessage(participants, participant) match {
+              case Left(error) => Future.successful(Left(error))
+              case Right(emailRequest) =>
+                emailConnector.send(emailRequest).flatMap {
+                  case Left(error) => Future.successful(Left(error))
+                  case Right(_)    => Future.successful(Right(()))
+                }
+            }
+          case _ => Future.successful(Left(StoreError("Message not create", None)))
```

Comment:
move both store errors repo getConverasationParticipants and repo.addMessageToConversation.